[DOCS-1608] Update Calico for Windows docs for HPC (#984)
tigera/docs: calico/getting-started/kubernetes/windows-calico/quickstart.mdx

| ServiceCidr        | Service IP range of the Kubernetes cluster. Not required for most managed Kubernetes clusters. Note: EKS has non-default value.                                                                                                                                                                  | 10.96.0.0/12 |
| DNSServerIPs       | Comma-delimited list of DNS service IPs used by Windows pod. Not required for most managed Kubernetes clusters. Note: EKS has a non-default value.                                                                                                                                               | 10.96.0.10   |
| CalicoBackend      | Calico backend network type (`vxlan` or `bgp`). If the value is an empty string (default), backend network type is auto detected.                                                                                                                                                                | ""           |

### Install {{prodnameWindows}} using HostProcess containers

> With Kubernetes v1.22, there is a new Windows container type called "HostProcess containers" (HPC) that can run directly on the host with access to the host network namespace,

Reviewer: ```suggestion
With Kubernetes v1.22, there is a new Windows container type called **HostProcess containers (HPC)** that can run directly on the host with access to the host network namespace,
```

Author: Done, thanks!

- Kubernetes v1.22+

> - HostProcess containers support enabled: for v1.22, HostProcess containers support has to be [enabled](https://kubernetes.io/docs/tasks/configure-pod-container/create-hostprocess-pod/#before-you-begin). For Kubernetes v1.23+, HostProcess containers are enabled by default.

Reviewer: ```suggestion
- Enable HostProcess containers support. For Kubernetes v1.22, see [enabled](https://kubernetes.io/docs/tasks/configure-pod-container/create-hostprocess-pod/#before-you-begin). For Kubernetes v1.23+, HostProcess containers are enabled by default.
```

Author: Done, please take a look!

kubeadm token create --print-join-command
```

- Edit the join command by updating the kubeadm.exe path to `c:\k\kubeadm.exe`.

> - If using a kubernetes version below v1.25, also append `--cri-socket "npipe:////./pipe/containerd-containerd"` to the join command.

Reviewer: ```suggestion
- If using a Kubernetes version v1.25 or earlier, append `--cri-socket "npipe:////./pipe/containerd-containerd"` to the join command.
```

Author: Done (almost, since this would change the meaning (we want to mean versions before v1.25, not including it)

>   An example join command:

Reviewer: At the end of the Requirements section, you have the bullet about "The Windows nodes have joined the cluster".
It would be easier to follow if you moved the two items about joining the cluster right after that bullet. And push the one about "To install containerd on the Windows node and configure the containerd service" after.

Author: Hm, I don't think that would work. I'm not sure if that's clear enough (appreciate suggestions to do so), but all of these steps are part of (or prereqs to) joining the cluster...

- Run the join command on the Windows node. Shortly after it completes successfully, the Windows node will appear in `kubectl get nodes`.
  The new node's status will be NotReady since the Calico CNI has not yet been installed.

#### Migrating from {{prodnameWindows}} installed manually to an HPC installation

> If your Windows nodes already have {{prodnameWindows}} installed using the manual installation method, you can continue this quickstart guide

Reviewer: ```suggestion
The HPC installation in this guide is operator-based and will automatically migrate your Windows nodes that have {{prodnameWindows}} installed using the manual installation method. This installation will uninstall any existing {{prodnameWindows}} services, and overwrite the {{prodnameWindows}} installation files with those included in the `calico/cni-windows` and `calico/node-windows` images. If `kubelet` was installed using `{{rootDirWindows}}\kubernetes\install-kube-services.ps1`, this service will remain installed. If the `kube-proxy` service was also installed using `{{rootDirWindows}}\kubernetes\install-kube-services.ps1`, it will be stopped and uninstalled by the `calico-node-windows` HPC initContainers. It is recommended to run `kube-proxy-windows` as hostprocess container images using the [manifest provided in the kubernetes-sigs sig-windows-tools repository](https://raw.githubusercontent.com/kubernetes-sigs/sig-windows-tools/master/hostprocess/calico/kube-proxy/kube-proxy.yml).
```

Author: Done, thanks!
